tket/src/Architecture/Architecture.cpp

```diff
@@ -18,37 +18,26 @@
 #include <unordered_set>
 #include <vector>
 
-#include "Circuit/Conditional.hpp"
 #include "Graphs/ArticulationPoints.hpp"
 #include "Utils/Json.hpp"
 #include "Utils/UnitID.hpp"
 
 namespace tket {
 
-// basic implementation that works off same prior assumptions
-// TODO: Update this for more mature systems of multi-qubit gates
-bool Architecture::valid_operation(
-    const Op_ptr& op, const std::vector<Node>& uids) const {
-  if (op->get_desc().is_box() ||
-      (op->get_type() == OpType::Conditional &&
-       static_cast<const Conditional&>(*op).get_op()->get_desc().is_box())) {
-    return false;
+bool Architecture::valid_operation(const std::vector<Node>& uids) const {
+  for (Node n : uids) {
+    if (!this->node_exists(Node(n))) return false;
   }
   if (uids.size() == 1) {
```

Comment:
Now this is just for Nodes and not OpTypes, and is not used in Routing, I guess this should check if the node is in the architecutre before returning true? its a different circumstance than to during mapping

Reply:
Yes it should, that is already added in the for loop at the beginning of the function.